utils/log_parser.py

```diff
@@ -0,0 +1,311 @@
+# Copyright (c) 2017-present, Facebook, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import re
+
+# Maintaining all nodes
+NODES_MAP = {}
+
+# Maintaining all edges
+EDGES_MAP = {}
+
+# Scope stack
+SCOPE_STACK = []
+
+# Critical Phases
+CRITICAL_PHASES = ["lower", "optimize", "fold", "optimizeFunction",
```

Comment:
This list of phases needs to be kept in sync with the Glow compiler code (eg. if somebody adds a new phase they need to update this). Can we remove it somehow?

Perhaps if a list of phases is not provided on the command line we can build a list of phases that were encountered parsing the log?